app.js

```diff
@@ -1,55 +1,49 @@
-const computerChoiceDisplay = document.getElementById("computer-choice")
-const userChoiceDisplay = document.getElementById("user-choice")
-const resultDisplay = document.getElementById("result")
-const possibleChoices = document.querySelectorAll("button")
-let userChoice
-let computerChoice
-let result
+const resultMap = {
+    'rock': {
+        'paper': false,
+        'scissors': true,
+    },
+    'paper': {
+        'rock': true,
+        'scissors': false,
+    },
+    'scissors': {
+        'rock': false,
+        'paper': true
+    }
+};
 
-possibleChoices.forEach(possibleChoice => possibleChoice.addEventListener("click", (e) => {
-    userChoice = e.target.id
-    userChoiceDisplay.innerHTML = userChoice
-    generateComputerChoice()
-    getResult()
-}))
+const generateComputerChoice = () => {
+    const randomNumber = Math.floor(Math.random() * possibleChoices.length) + 1;
 
-function generateComputerChoice() {
-    const randomNumber = Math.floor(Math.random() * possibleChoices.length) + 1
+    switch (randomNumber) {
+        case 1:
+            return 'rock';
+        case 2:
+            return 'scissors';
+        case 3:
+            return 'paper';
+    };
```

Comment:
Switches are great - they're great for when you want to do something based on the state of something else that could be many different values. They're also actually a bit faster than `if / else`s - but I'd only use them if we're looking at three or more things.

I'm also returning out here, as I want my function to *only* do the thinking. I can then decide whatever I want with the computer choice string before setting `innerHTML` to it - heck, I might want to console log it or something.